executor/aggfuncs/func_max_min.go

```diff
@@ -0,0 +1,299 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package aggfuncs
+
+import (
+	"github.com/juju/errors"
+	"github.com/pingcap/tidb/sessionctx"
+	"github.com/pingcap/tidb/types"
+	"github.com/pingcap/tidb/util/chunk"
+)
+
+type partialResult4MaxMinInt struct {
+	val int64
+	// isNull is used to indicates:
+	// 1. whether the partial result is the initialization value which should not be compared during evaluation;
+	// 2. whether all the values of arg are all null, if so, we should return null as the default value for MAX/MIN.
+	isNull bool
+}
+
+type partialResult4MaxMinUint struct {
+	val    uint64
+	isNull bool
+}
+
+type partialResult4MaxMinDecimal struct {
+	val    *types.MyDecimal
```

Comment:
The input value may be changed when child operator reuses its input `Chunk`, we should store the copied value to avoid this issue, change `val *types.MyDecimal` to `val types.MyDecimal`.